src/ack_manager.rs

```diff
@@ -56,11 +60,15 @@ impl AckManager {
         }
     }
 
-    // Handles a received ack (removes the corresponding message from the list of
-    // pending ones, and remembers that we have received this ack).
+    /// Handles a received ack (removes the corresponding message from the list of
+    /// pending ones, and remembers that we have received this ack).
     pub fn receive(&mut self, ack: Ack) {
-        let _ = self.pending.remove(&ack);
-        let _ = self.received.insert(&ack);
+        let _ack = self.pending
+            .remove(&ack)
+            .ok_or(debug!("received and Ack we were not expecting : {}", ack));
+        // TODO - Should this insert an ack we were not expecting ??
```

Comment:
Yes. It can happen that the rest of our group already sent the message and we're receiving the acknowledgement before even considering to send it ourselves. I'm not even sure we should log it, but we can try it out and see how noisy it is.